graphs-lab/src/main/kotlin/graphs_lab/algs/DijkstraAlgorithm.kt

```diff
@@ -0,0 +1,170 @@
+	 * to this set beginning with the given identifier.
+	 */
+	private val table = mutableMapOf<I, Pair<I?, Double>>()
+
+	/**
+	 * [statusTable] the status of [table] equals a pair that consist of the given identifier,
+	 * from which come out, and a boolean variable that shows whether the algorithm was
+	 * processed without exceptions.
+	 */
+	private var statusTable: Pair<I, Boolean>? = null
+
+	/**
+	 * Returns the shortest path from a source vertex to a target vertex.
+	 *
+	 * @param idSource the identifier of the source vertex
+	 * @param idTarget the identifier of the target vertex
+	 * @return the path equals a list of the identifier of a vertex in the order of movement
+	 * @throws IllegalArgumentException if a weight is negative number or [idSource] or [idTarget]
+	 * is not contained in the graph
+	 * @throws IllegalStateException if there is an appeal to a non-existent identifier
+	 */
+	fun getPath(idSource: I, idTarget: I): MutableList<I> {
+		val path = mutableListOf<I>()
+
+		if (statusTable == null || checkAndGetFirst(statusTable) != idSource) {
+			getAllPaths(idSource)
+		}
+		check(checkAndGetSecond(statusTable)) { "All paths do not build." }
+		require(table[idTarget] != null) { "The target vertex is not contained." }
+		if (table[idTarget]?.first == null) return path
+
+		// Assembles the path in the reversed order of movement
+		var idCurrent = idTarget
+		path.add(idCurrent)
+		while (idCurrent != idSource) {
+			idCurrent = checkAndGetFirst(table[idCurrent])
+			path.add(idCurrent)
+		}
+
+		return path.asReversed()
+	}
+
+	/**
+	 * Returns a weight of the shortest path from a source vertex to a target vertex.
+	 *
+	 * @param idSource the identifier of the source vertex
+	 * @param idTarget the identifier of the target vertex
+	 * @return a weight equals the sum of the all weights of edges on the path from the given identifier
+	 * @throws IllegalArgumentException if a weight is negative number or [idSource] or [idTarget]
+	 * is not contained in the graph
+	 * @throws IllegalStateException if there is an appeal to a non-existent identifier
+	 */
+	fun getWeightPath(idSource: I, idTarget: I): Double {
+		if (statusTable == null || checkAndGetFirst(statusTable) != idSource) {
+			getAllPaths(idSource)
+		}
+		check(checkAndGetSecond(statusTable)) { "All paths do not build." }
+		require(table[idTarget] != null) { "The target vertex is not contained." }
+		if (table[idTarget]?.first == null) return Double.POSITIVE_INFINITY
+
+		return checkAndGetSecond(table[idTarget])
+	}
+
+	/**
+	 * @param idSource the identifier of the source vertex
+	 * @return [table]
+	 * @throws IllegalArgumentException if a weight is negative number or [idSource] is not contained in the graph
+	 * @throws IllegalStateException if there is an appeal to a non-existent identifier
+	 */
+	fun getAllPaths(idSource: I): MutableMap<I, Pair<I?, Double>> {
```

Comment:
Название функции предполагает, что пользователю вернут пути ко всем достижимым вершинам, а здесь возвращается стоимость минимального пути. Что-то мне подсказывает, что что-то не так.
Возможно имеет эту функцию сделать `private` и переименовать. Также добавить такую функцию:
```kotlin
	fun getAllPaths(idSource: I): Map<I, List<I>> {
		val shortestPaths = mutableMapOf<I, List<I>>()
		for (idVertex in graph.idVertices) {
			shortestPaths[idVertex] = getPath(idSource, idVertex)
		}
		return shortestPaths
	}
```
Итого:
```suggestion
	fun getAllPaths(idSource: I): Map<I, List<I>> {
		val shortestPaths = mutableMapOf<I, List<I>>()
		for (idVertex in graph.idVertices) {
			shortestPaths[idVertex] = getPath(idSource, idVertex)
		}
		return shortestPaths
	}
	
	private fun getAllMinimalParent(idSource: I): Map<I, Pair<I?, Double>>
```